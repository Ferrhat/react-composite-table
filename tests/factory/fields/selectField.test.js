import React from 'react';
import {shallow} from 'enzyme';
import SelectField from "../../../lib/factory/fields/SelectField";


describe('SelectField', () => {
    const mockOnChange = jest.fn();
    const selectField = shallow(<SelectField onChange={mockOnChange} options={[]} name={'testName'} filterableProperty={'testValue'} />);

    it('renders properly', () => {
        expect(selectField).toMatchSnapshot();
    });

    it('contains a connected Select component', () => {
        expect(selectField.find('Select').length).toEqual(1);
    });

    it('should trigger a change event', () => {
        const selection = {
            value: 'testValue',
        };
        selectField.find('Select').simulate('change', selection);
        expect(mockOnChange).toBeCalledWith('testName', 'testValue', selection);
    });
});
