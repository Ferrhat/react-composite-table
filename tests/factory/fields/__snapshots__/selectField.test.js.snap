// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectField renders properly 1`] = `
<Select
  arrowRenderer={[Function]}
  autosize={true}
  backspaceRemoves={false}
  backspaceToRemoveMessage="Press backspace to remove {label}"
  clearAllText="Clear all"
  clearRenderer={[Function]}
  clearValueText="Clear value"
  clearable={true}
  closeOnSelect={true}
  deleteRemoves={true}
  delimiter=","
  disabled={false}
  escapeClearsValue={true}
  filterOptions={[Function]}
  ignoreAccents={true}
  ignoreCase={true}
  inputProps={Object {}}
  isLoading={false}
  joinValues={false}
  labelKey="label"
  matchPos="any"
  matchProp="any"
  menuBuffer={0}
  menuRenderer={[Function]}
  multi={true}
  name="testName"
  noResultsText="No results found"
  onBlurResetsInput={true}
  onChange={[Function]}
  onCloseResetsInput={true}
  onSelectResetsInput={true}
  openOnClick={true}
  optionComponent={[Function]}
  options={Array []}
  pageSize={5}
  placeholder="Select..."
  removeSelected={true}
  required={false}
  rtl={false}
  scrollMenuIntoView={true}
  searchable={true}
  simpleValue={false}
  style={
    Object {
      "height": "100%",
    }
  }
  tabSelectsValue={true}
  trimFilter={true}
  value={null}
  valueComponent={[Function]}
  valueKey="value"
/>
`;
